Do not query interface's self_param_id unless defined when importing

gysddn opened the pull request:
Querying a local constant with an invalid instruction triggers a crash, this will prevent self_param_id to be used unless it is defined.

Closes #4071 and #4080

This will only fix the crash. The scenario where a declaration-only interface is imported then defined still gives an error message, but it won't crash this time.

gysddn committed:
Do not query interface's self_param_id unless defined when importing

Querying a local constant with an invalid instruction triggers a crash,
this will prevent self_param_id to be used unless it is defined.

Fixes #4071 and #4080

zygoloid reviewed (commented):
Changes look good, but please can you also add a test?

gysddn committed:
Add tests for interface decl imprort and interface redeclaration

- Covers the test cases in #4071 and #4080
- Importing declaration works without an issue
- Importing, then defining the interface throws a duplicate declaration
issue since it is not yet implemented.

gysddn commented:
Added two cases to cover both issues 👍🏻

zygoloid reviewed (approved):
Thank you!

gysddn committed:
Merge branch 'trunk' into fix4071
